scripts/jenkins/Makefile.jenkins

```diff
@@ -347,17 +319,17 @@ test-junit-7-smoke-jenkins:
 	@$(MAKE) -f $(THIS_FILE) test-junit-7-smoke
 
 test-junit-10-smoke:
-	DOONLY=$$(head -n 1 tests/doOnlyJunitSmokeTestList) ./gradlew h2o-core:test h2o-algos:test $$ADDITIONAL_GRADLE_OPTS
+	ADDITIONAL_TEST_JVM_OPTS="-Dsys.ai.h2o.debug.allowJavaVersions=10" DOONLY=$$(head -n 1 tests/doOnlyJunitSmokeTestList) ./gradlew h2o-core:test h2o-algos:test $$ADDITIONAL_GRADLE_OPTS
```

Comment:
Why is this needed? To make sure we have the right java in the docker image?

Reply:
honestly no idea what it does, was missing in some of the targets so I added it to make things consistent

and yes, this will check that the tests are actually running on the java version we think they are running on, makes sense

Reply:
ok, it should be removed then

I checked and it adds a user-defined version to the list of supported versions

the reason why it was there was that someone forgot to remove it, it only should be there when you are eg. on a release branch and want to test against a new java and don't want to change it in the code (you don't know it works yet)

so you add this property for say Java 15, merge the PR and let the nightlies confirm there is no issue

then you go back to the code, modify the version check in java and remove this property - which is not needed anymore

Reply:
ok, will do